### DISCUSSION TITLE
Delay in weather observations

### DISCUSSION THREAD

**Discussion by jamesharr** [weather-gov/api#332] [Q&A]
I'm looking at weather stations in my area (specifically KMLE, KOMA) and it seems like observations are delayed anywhere from 30 minutes. There have been times where I've seen readings several hours old.

Is there any guidance on how fresh I should expect that data to be? Or is there any guidance on what that's up to? IE: is it the local weather station that dictates that? Is it a universal thing?

**Comment by StephenClouse** [comment]
We upgraded the API over the last week and we are still investigating what may be a bug that is causing at least some of this. Please bear with us for a bit.

**Comment by Ahowe042** [comment]
I would like to follow up with this by mentioning that I am also experiencing strange behavior with the Observations API route. 

I have attached two screenshots here, one from a CSharp app (1st screenshot), and the other from Postman (2nd screenshot). Both of these requests were executed several times within seconds of each other, and the API route was identical (despite the 1st screenshot showing ID ending in a timestamp rather than the "latest" keyword). 

The Postman request always returned the most up-to-date data, while the CSharp-based request always returned data that was several hours old.

![Screen Shot 2020-06-23 at 8 12 32 PM](https://user-images.githubusercontent.com/7587250/85481342-a61b9e00-b58f-11ea-994a-be7e44fe5334.png)
![Screen Shot 2020-06-23 at 8 12 47 PM](https://user-images.githubusercontent.com/7587250/85481345-a74ccb00-b58f-11ea-8b9c-c3d1bd682c32.png)

**Comment by jamesharr** [comment]
Is there any information or aide I can provide that would be useful?

**Comment by scadergit** [comment]
The issue that @Ahowe042 detailed has a patch that is being deployed to QA this morning, likely production shortly after.

A short delay up to 20 minutes for observations is a known issue with the upstream source for the API. We are working with that source to improve that latency, but that won't change with this patch.
